### ISSUE TITLE
crafting priority upgrade

### ISSUE THREAD

**Issue by loordgek** [RS485/LogisticsPipes#826]
the problem w/ crafting module when you request a tesseract frame the induction smelter recipes get mixed up 
this upgrade is preventing that by only allowing one crafting module to craft the others not 

![craft](https://cloud.githubusercontent.com/assets/4789835/10563487/2616af62-758c-11e5-9010-bb99046046eb.jpg)

**Comment by SonOfTheStars** [comment]
Isnt that effect mitigated by using MK3 Crafters? never had such a problem.
Plus you could just use 2 Different Induction Smelters for the sake of being faster through parallel crafting
I think a crafting priority upgrade isn't really neccessary...
It creates a bit of a struggle. Easy is not allways good

**Comment by hron84** [comment]
I also vote for using two different induction smelters, because in this way you can feed one constantly with pyrotheum dust and in recipes you can just set e.g. enderium dust -> enderium ingot, that makes crafting faster and better and also causes less lag if you use BC pipes for transport items.

**Comment by davboecki** [comment]
If I understand you correctly you are using one crafting machine with several crafting modules. People doing this was one of the reasons I didn't want to make crafitng modules in the first place. LP is not designed to handle several recipies on one machine of this kind. Machines with input inventories would work, but machines with only limited input slots WILL destroy your crafting process. That's just not the way LP was designed.

**Comment by loordgek** [comment]
@davboecki  but machines with only limited input slots WILL destroy your crafting process
can you explain that 
and maybe reopen this

**Comment by davboecki** [comment]
LP's crafting pipes don't have a sense of order when it comes to crafting tasks. They just attempt to extract every item they expect to be there and when it is they take it. If the item is not there they try the next item ordered and requeue the failed item. Same goes for the provider pipe sending to the crafting pipe. It doesn't care what task the crafting pipe currently has, because they can change every tick, so the provider pipe sends the item to the crafting pipe directly. All the tasks are calculated inside the RequestTree and then handed of to the crafting and provider pipes, after that there is no reference to that RequestTree anymore just single tasks that need to be fullfilled. So orders will get switched around at random causing all type of lockups when you use several crafting modules inside one chassis pipe on one machine.

**Comment by loordgek** [comment]
can you make that the crafting modules are "talking" to each other.
need to go to bed more will follow later

**Comment by bziemons** [comment]
It would make no sense having the crafting modules talking to each other. What would be a solution is having something like a lock for each pipe. A crafting module in a chassis pipe would then lock the chassis pipe until it is done with it's task. The next crafting module can try to lock the pipe, but fails until the other module is done. It's like threads trying to access one resource to write, basically.

**Comment by SonOfTheStars** [comment]
Meh!
Id like to not enable "Locks" on  Modules.
Making it so easy for Users, you could just advertise AE2...
Im strickly against the "One Machine for everything", it makes gameplay boring.

Users should be forced to resolve this on their own.

**Comment by loordgek** [comment]
i like the crafting wall but not a row off Machines

**Comment by hron84** [comment]
And how about a wall of Machines? :-)

**Comment by loordgek** [comment]
no

**Comment by SonOfTheStars** [comment]
Why the strict "no!"? Makes no sence. If you can do a Wall-o-Crafting, you can manage to do 2-3 machines of the same Kind. because thats all you'll even gonna need!

For Example with the Induction Smelter:
If you have 3 of them and Spread the recips across them you can effectively negate the Effects you want this Upgrade for.
Its all about figuring out how your Crafting Processes for, eg. making a tesseract get issued and develop an order, so you can ensure no materials that are based on another material get produced in the same machine.

Your problem is dealt with and youve had a good time figuring it out! 

EZ Modules are for the Weak!

**Comment by hron84** [comment]
To be really honest, AE2 expects nearly same thing from you, as it needs duplicated recipes (and machines) to enable parallel crafting in most cases.

**Comment by SonOfTheStars** [comment]
It even needs Parralell machines for stuff like Enderium sometimes...

**Comment by hron84** [comment]
Enderium is not needs, but Tesserracts... :smile:

**Comment by SonOfTheStars** [comment]
Still, Im strictly against "One Machine for X >= 3" Tasks at once. Plus Parallel Crafting makes your Crafting alot faster!

Id even be fine with "1 Machine for X = 1" Tasks at Once, if that wherent quite laggy and a pain in the ass for things Like IC2.

**Comment by bziemons** [comment]
Well, opinions are opinions, but I'd still like to say something about yours @Bitterholz.

> Id like to not enable "Locks" on Modules.

I was not talking about generally having locks on modules, I was specifically talking about one lock for crafting tasks per chassis pipe.

> Making it so easy for Users, you could just advertise AE2...

This comment makes no sense. If anyone wants to use AE, they should do. You are just being against AE in general, but that does not belong here.

> Im strickly against the "One Machine for everything", it makes gameplay boring.

There is no machine for everything in LP and this comment makes no sense. LP had modules forever and chassis pipes have limitations as well as the modules do.

> Users should be forced to resolve this on their own.

Well, they can't. We should disallow putting multiple crafting modules in a chassis pipe then, because the way it works right now really is buggy and uncontrolled.

**Comment by hron84** [comment]
@theZorro266 I'd not like to protect @Bitterholz from the comments, but I'd like to say: Induction Smelters from TE is make a headache for other automation stuffs too, not only for LP, as AE cannot handle them correctly too. I still remember to AS1 where I made multiple IS for tasks (and I still weren't too happy with them). Mass-requesting is still an issue for IS and you - as a player - have to solve it your own. Regardless if LP crafting is buggy or not.

**Comment by bziemons** [comment]
@hron84 maybe I get you wrong, but I do not know what that has to do with what I suggested. It solely seems to point at the comment that "users should resolve this on their own" and of course that could be the way we are going, but the current behavior seems "buggy and uncontrolled".

I opened a new enhancement/bug issue, because I am not sure what it will be, at #843. So any discussion on that can be held there. The first idea in this issue is not good, as stated by @davboecki.
